app/actions/local/channel.test.ts

```diff
@@ -0,0 +1,468 @@
+import ServerDataOperator from '@database/operator/server_data_operator';
+import {dismissAllModalsAndPopToRoot, dismissAllModalsAndPopToScreen} from '@screens/navigation';
+
+import {switchToChannel} from './channel';
+
+let mockIsTablet: jest.Mock;
+const now = new Date('2020-01-01').getTime();
+
+jest.mock('@screens/navigation', () => {
+    const original = jest.requireActual('@screens/navigation');
+    return {
+        ...original,
+        dismissAllModalsAndPopToScreen: jest.fn(),
+        dismissAllModalsAndPopToRoot: jest.fn(),
+    };
+});
+
+jest.mock('@utils/helpers', () => {
+    const original = jest.requireActual('@utils/helpers');
+    mockIsTablet = jest.fn(() => false);
+    return {
+        ...original,
+        isTablet: mockIsTablet,
+    };
+});
+
+describe('switchToChannel', () => {
+    let operator: ServerDataOperator;
+    let spyNow: jest.SpyInstance;
+    const serverUrl = 'baseHandler.test.com';
+    const channelId = 'id1';
+    const teamId = 'tId1';
+    const team: Team = {
+        id: teamId,
+    } as Team;
+    const channel: Channel = {
+        id: channelId,
+        team_id: teamId,
+        total_msg_count: 0,
+    } as Channel;
+    const channelMember: ChannelMembership = {
+        id: 'id',
+        channel_id: channelId,
+        msg_count: 0,
+    } as ChannelMembership;
+    beforeEach(async () => {
+        await DatabaseManager.init([serverUrl]);
+        operator = DatabaseManager.serverDatabases[serverUrl].operator;
+        spyNow = jest.spyOn(Date, 'now').mockImplementation(() => now);
+    });
+
+    afterEach(async () => {
+        await DatabaseManager.destroyServerDatabase(serverUrl);
+        spyNow.mockRestore();
+    });
+
+    it('handle not found database', async () => {
+        const listenerCallback = jest.fn();
+        const listener = DeviceEventEmitter.addListener(Navigation.NAVIGATION_HOME, listenerCallback);
+        const {error} = await switchToChannel('foo', 'channelId');
+        listener.remove();
+
+        expect(error).toBeTruthy();
+        expect(dismissAllModalsAndPopToScreen).toHaveBeenCalledTimes(0);
+        expect(dismissAllModalsAndPopToRoot).toHaveBeenCalledTimes(0);
+        expect(listenerCallback).toHaveBeenCalledTimes(0);
+    });
+
+    it('handle no member', async () => {
+        const listenerCallback = jest.fn();
+        const listener = DeviceEventEmitter.addListener(Navigation.NAVIGATION_HOME, listenerCallback);
+        const {models, error} = await switchToChannel(serverUrl, 'channelId');
+        listener.remove();
+
+        expect(error).toBeUndefined();
+        expect(models?.length).toBe(0);
+
+        const systemValues = await queryCommonSystemValues(operator.database);
+        const teamHistory = await queryTeamHistory(operator.database);
+        const channelHistory = await queryChannelHistory(operator.database, teamId);
+        const member = await queryMyChannel(operator.database, channelId);
```

Comment:
This block seems common across the tests and might be extracted to it's own function